Clarify that Baseline does not cover assistive technology (#519)
web-platform-dx/web-features: docs/baseline.md

  Many Baseline features will not achieve 100% user reach soon or perhaps ever.
  If a web developer needs to support a globally uncommon or discontinued browser (e.g., Internet Explorer 11), then the developer needs to know the specific limitations of that browser, not a broad overview of developers’ most commonly required browsers.
  Baseline can’t be both.

> * **Identify support in assistive technology.**
>   Baseline does not cover support in assistive technology (AT), which includes screen readers, screen magnifiers, voice control, and so on.
>   See also: [Future considerations](#future-considerations).

Reviewer: I find myself agreeing with @foolip's https://github.com/web-platform-dx/web-features/pull/519#pullrequestreview-1883062568, but also reading this text as in conflict with that understanding—or at very least, I don't know what I ought to do or not do with this non-goal.

Perhaps we can be explicit about the way in which this non-goal relates to developers' role in accessibility?

```suggestion
* **Identify support in assistive technology.**
  Baseline does not cover support for screen readers, screen magnifiers, voice control, and other assistive technology that browsers do not expose to developers and their applications.
  See also: [Future considerations](#future-considerations).
```

(I've also omitted the initialism, since we don't reuse this term elsewhere in the document.)

Author: Ah, I don't mind changing the text but I read *your* suggested update as in conflict with @foolip's comment ;)

Much of the "accessibility tree" that @foolip mentions as *in scope* is internal to browsers I believe, exposed to AT for sure, but not exposed to web developers and their applications. At least not until the [Accessibility Object Model](https://wicg.github.io/aom/spec/) specs make further progress. With that in mind, I'm reading your suggestion as saying that if the browser does not expose the correct information in the accessibility tree, then that's *out of scope* for Baseline.

Reviewer: Haha OK, I guess I don't know quite how to capture this line of thinking:

Baseline should care about the consistency of the core browser set to the extent that developers are offered a facility to control/manage/influence _browser_ behavior.

Things that ought to be out-of-scope:

- Plug-in, operating system, hardware, or non-core browser set user agent limitations (e.g., a bug in JAWS or VoiceOver, lack of OS support for a given Braille display, etc.)
- Inaccessible browser behavior required or permitted by specification, provided that behavior is consistent with other browsers (e.g., if browsers, as a group, are doing things that are _wrong_ but correct and consistent, then we ought to be able to say that developers can rely on that behavior, even if we hope that developers will make a judgment to not use those things).
- Subjective variation between browsers in terms of the user interaction model or quality (we'll see this in other areas, like variations on when and how browsers prompt users for permissions).

(I think the accessibility tree is a slightly special case here in that browsers—rather consistently—show developers the tree in developer tools, which makes a sort of implicit agreement with them about the way pages ought to alter the tree but I also think that's a specific conversation we could have, whether or not we adopt a change to the definition.)

Reviewer 2: @tidoust
> Much of the "accessibility tree" that @foolip mentions as in scope is internal to browsers I believe, exposed to AT for sure, but not exposed to web developers and their applications

The accessibility tree (which is itself an abstraction) is exposed to authors in dev tools and individual properties are exposed in assorted dev tools views. AOM is not directly related to this for today.

Author: Right @aardrian, but when I see "exposed to developer *and their applications*", I read interfaces, IDL, not dev tools views. That's what I'm reacting to here.

@ddbeck, perhaps dropping "and their applications" is enough to make things clear, or at least clear enough.

Reviewer 3: What @ddbeck wrote matches my thinking. I mentioned the accessibility tree, but we don't need to focus too much on that. If  keyboard navigation doesn't work because of a browser bug, that also counts. Whereas an OS bug is out of scope for Baseline, but could of course be relevant to mention in BCD notes, and in web-features notes if we ever add that.

Reviewer: I don't think there's any loss of clarity (and certainly no gain) if we drop the "and their applications" bit:

```suggestion
* **Identify support in assistive technology.**
  Baseline does not cover support for screen readers, screen magnifiers, voice control, and other assistive technology that browsers do not expose to developers.
  See also: [Future considerations](#future-considerations).
```

>   Baseline should cover core accessibility support in browsers but it does not cover support in assistive tools (e.g., screen magnifiers, screen readers, alternative keyboards) that may be used along a browser to meet the requirements of people with disabilities.

Reviewer 2: Suggested re-write since this is a non-goal (and an aspirational _should_ seems weird here):

"Baseline will not cover support in assistive technology (AT), which includes screen readers, screen magnifiers, voice control, and so on."

  If a web developer needs to support a globally uncommon or discontinued browser (e.g., Internet Explorer 11), then the developer needs to know the specific limitations of that browser, not a broad overview of developers’ most commonly required browsers.
  Baseline can’t be both.
* **Identify support in assistive technology.**

>   Baseline does not cover support for screen readers, screen magnifiers, voice control, and other assistive technology that browsers do not expose to developers.

Reviewer 3: I might clarify it like this:

```suggestion
  Baseline does not cover support for screen readers, screen magnifiers, voice control, and other assistive technology that is not built into browsers.
```

I would also make it "Identify support in assistive technology outside of browsers" above.

Reviewer 4: +1 to the suggested change.

* Upcoming (e.g., not yet shipped in all browsers)
* Progressive enhancement safe (i.e., limited penalties for support failures)
* Developer feedback requested

> * Buggy (e.g., supported but in ways that are surprising and semi-interoperable, not exposed to platform accessibility APIs (AAPIs) as specified)

Reviewer 5: I think I'd make accessible a separate bullet point, since something could be fully interoperable for non-AT users and fully unsupported in AT (I hope we don't have features in that state, but I still think it makes sense to make "accessible" a top level item rather than a sub-case of generally buggy).

Author: Done. To be clear, I don't necessarily think that Baseline should have a dedicated state for that. I was more using the list to make explicit that exposure to platform accessibility APIs is one of the areas that should be taken into account once we have enough support data.

* Progressive enhancement safe (i.e., limited penalties for support failures)
* Developer feedback requested
* Buggy (e.g., supported but in ways that are surprising and semi-interoperable)

> * Buggy or unsupported in assistive tools

Reviewer 2: Because this is a future consideration and Baseline is not going to add AT support, I suggest something a bit more straightforward as a future goal (since it keeps the scope strictly with browsers and does not require AT):

"Confirm if features are exposed to platform accessibility APIs (AAPIs) as specified."

> * Not exposed to platform accessibility APIs (AAPIs) as specified

Reviewer 3: To use the exact wording from above, now that it's gotten a lot of polish:

```suggestion
* Support in assistive technology that is not built into browsers.
```
